docs/src/index.md

```diff
@@ -2,13 +2,228 @@
+  [Transducers.jl](https://github.com/JuliaFolds/Transducers.jl)
+  framework.
+
+- Pipelines can be reapplied to unseen "test" data using the same cache
+  (e.g. constants) fitted with "training" data. For example, a `ZScore`
+  relies on "fitting" `μ` and `σ` once at training time.
+
+## Rationale
+
+A common task in statistics and machine learning consists of transforming
+the variables of a problem to achieve better convergence or to apply methods
+that rely on multivariate Gaussian distributions. This process can be quite
+tedious to implement by hand and very error-prone. We provide a consistent
+and clean API to combine statistical transforms into pipelines.
+
+*Although most transforms discussed here come from the statistical domain,
+our long term vision is more ambitious. We aim to provide a complete
+user experience with fully-featured pipelines that include standardization
+of column names, imputation of missing data, and more.*
+
+## Installation
+
+Get the latest stable release with Julia's package manager:
+
+```julia
+] add TableTransforms
+```
+
+## Usage
+
+Below is a quick example with simple transforms:
+
+```julia
+using TableTransforms
+using PairPlots
+
+# example table from PairPlots.jl
+N = 100_000
+a = [2randn(N÷2) .+ 6; randn(N÷2)]
+b = [3randn(N÷2); 2randn(N÷2)]
+c = randn(N)
+d = c .+ 0.6randn(N)
+table = (;a, b, c, d)
+
+# corner plot of original table
+table |> corner
+```
+![original](assets/original.png)
+
+```julia
+# convert to PCA scores
+table |> PCA() |> corner
+```
+![pca](assets/pca.png)
+
+```julia
+# convert to any Distributions.jl
+table |> Quantile(Normal()) |> corner
+```
+![quantile](assets/quantile.png)
+
+Below is a more sophisticated example with a pipeline that has
+two parallel branches. The tables produced by these two branches
+are concatenated horizontally in the final table:
+```julia
+# create a transform pipeline
+f1 = ZScore()
+f2 = Scale()
+f3 = Quantile()
+f4 = Functional(cos)
+f5 = Interquartile()
+pipeline = (f1 → f2 → f3) ⊔ (f4 → f5)
+
+# feed data into the pipeline
+table |> pipeline |> corner
+```
+![pipeline](assets/pipeline.png)
+
+To revert a pipeline or single transform, use the `apply` and `revert`
+functions instead:
+
+```julia
+# apply transform and save cache to revert later
+newtable, cache = apply(pipeline, table)
+
+# perform additional modeling with newtable
+# newtable = ...
+
+# revert pipeline when done with modeling
+original = revert(pipeline, newtable, cache)
 ```
 
-```@autodocs
-Modules = [TableTransforms]
+Finally, it is sometimes useful to `reapply` a transform that was
+"fitted" with training data to unseen test data. In this case, the
+cache from a previous `apply` call is used:
+
+```julia
+# ZScore transform "fits" μ and σ using training data
+newtable, cache = apply(ZScore(), traintable)
+
+# we can reuse the same values of μ and σ with test data
+newtable = reapply(ZScore(), testtable, cache)
 ```
+
+## Available transforms
```

Comment:
We will have separate pages in the documentation for this section and all remaining sections. Can you create a separate page with a list of docstrings instead? We can then create a header for each transform so that users can quickly click on the name to jump to the docstring.